app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,30 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def update_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is present : return as it is
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
+                                        and updates metadata with response_id
+
+
+        """
+        if data.get("response_id"):
+            return data
+        eq_id = data.get("eq_id")
+        form_type = data.get("form_type")
+        ru_ref = data.get("ru_ref")
+        collection_exercise_sid = data.get("collection_exercise_sid")
+        if eq_id and form_type:
+            response_id = f"{ru_ref}{collection_exercise_sid}{eq_id}{form_type}"
+            data["response_id"] = response_id
+            return data
+
+        raise ValidationError("Either eq_id or form_type is missing")
```

Comment:
Minor, can we quote the claim name like the schema name one. Also, we should say why we need it as it is useful for debugging and upstream. As in, it is missing but required when response_id isn't provided. Something like that?.